sycl/source/accessor.cpp

```diff
@@ -119,19 +119,12 @@ const sycl::range<3> &LocalAccessorBaseHost::getSize() const {
   return impl->MSize;
 }
 void *LocalAccessorBaseHost::getPtr() {
-  // Const cast this in order to call the const getPtr.
-  return const_cast<const LocalAccessorBaseHost *>(this)->getPtr();
+  // Must not be called.
```

Comment:
For the sake of debugging, in case this doesn't hold, could we have an `assert` here that always fails?

Reply:
We have error handling in the base classes before/instead "delegating" to here.

Reply:
I'm not sure I follow. The reason I would like an assert here is to avoid us unintentionally adding code to the runtime that call these functions, be it directly or through other functions (like `getQualifiedPtr()`.)

Reply:
asserts won't be useful, because library is built in release mode. Added `abort`, even though I still don't see much value here. Ideally, these shouldn't exist at all, but I can't remove them as that would remove a symbol.

Reply:
I totally agree, these should be removed ASAP, but the `assert` would be there to ensure us developers don't do anything we shouldn't do. It may not be a huge amount of value, but it's practically free and would make it easier for the runtime dev to understand why things are crashing over a potential nullptr-deref. I can live with abort too though.

Reply:
The abort was hit by some MKL tests, when running on Windows, but not on Linux?? See CMPLRLLVM-65833